jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/util/CodeWhispererFileContextProvider.kt

```diff
@@ -282,5 +278,27 @@ class DefaultCodeWhispererFileContextProvider(private val project: Project) : Fi
 
     companion object {
         private val LOG = getLogger<DefaultCodeWhispererFileContextProvider>()
+
+        fun shouldFetchUtgContext(language: CodeWhispererProgrammingLanguage, userGroup: CodeWhispererUserGroup): Boolean? {
+            if (!language.isUTGSupported()) {
+                return null
+            }
+
+            return when (language) {
+                is CodeWhispererJava -> true
+                else -> userGroup == CodeWhispererUserGroup.CrossFile
+            }
+        }
+
+        fun shouldFetchCrossfileContext(language: CodeWhispererProgrammingLanguage, userGroup: CodeWhispererUserGroup): Boolean? {
+            if (!language.isSupplementalContextSupported()) {
+                return null
+            }
+
+            return when (language) {
+                is CodeWhispererJava -> true
+                else -> userGroup == CodeWhispererUserGroup.CrossFile
+            }
+        }
```

Comment:
should return `null` if the language is not supported by given feature
should return `false` if the language is supported but userGroup is `Control`